### PR TITLE
Update gcp & azure examples to go modules

### PR DESCRIPTION
**Pull request by EvanBoyle**
Some of them had code errors that I fixed. I ran a `pulumi preview` on all of them after the update to verify.

### CODE REVIEW

**Comment by EvanBoyle** [comment]
fixes https://github.com/pulumi/pulumi/issues/3817

File: azure-go-webserver-component/main.go [hunk 1]
```diff
@@ -41,6 +41,14 @@ func main() {
 			},
 		})
 
+		subnetID := network.Subnets.Index(pulumi.Int(0)).Id().ApplyT(func(val interface{}) (string, error) {
+			v := val.(*string)
```
Comment (lukehoban) [inline azure-go-webserver-component/main.go]: Total nit, but can this not be `func(v *string)`?

Comment (EvanBoyle) [reply azure-go-webserver-component/main.go]: Good catch. Updated.

**Comment by EvanBoyle** [comment]
Unrelated build failure. `TestAccPacketPyWebserver` has been failing in the master cron build for some time.

**Pull request merged**